### PR TITLE
test: Do not require netpols in 'waitNextPolicyRevisions()'

### PR DESCRIPTION
**Pull request by jrajahalme**
'waitNextPolicyRevisions()' currently returns 'true' when no k8s
network policies are applied, bypassing the Cilium agent policy
revision wait in this case. As our tests typically (never?) have no
NPs applied, we have not actually waited for CNP or CCNP changes to
take place in all Cilium PODs before proceeding with the tests. This
may have caused CI flakes.

Fix this by removing the code that checks for the presence of NPs.

Reported-by: Paul Chaignon <paul@cilium.io>
Signed-off-by: Jarno Rajahalme <jarno@isovalent.com>

### CODE REVIEW

**Review by christarazi — COMMENTED** [review]
(Meant to hit approve, can't delete this comment)

**Review by christarazi — APPROVED** [review]
:open_mouth:

File: test/helpers/kubectl.go [hunk 1]
```diff
@@ -2918,43 +2918,23 @@ func (kub *Kubectl) getPodRevisions() (map[string]int, error) {
...
-		knp := kub.ExecShort(fmt.Sprintf("%s get --all-namespaces netpol -o jsonpath='%s'",
-			KubectlCmd, npFilter))
-		result := knp.ByLines()
-		if len(result) == 0 {
-			return true
-		}
-
-		for _, item := range result {
-			for ciliumPod, revision := range podRevisions {
-				if mustHavePolicy {
```
Comment (tklauser) [inline test/helpers/kubectl.go:2935]: It looks like `mustHavePolicy` is no longer used after this change. Can we remove it from `waitNextPolicyRevisions` arguments?

**Review by pchaigno — APPROVED** [review]

File: test/helpers/kubectl.go (hunk 1 above)
Comment (jrajahalme) [reply test/helpers/kubectl.go:2935]: Done :-)

**Review by tklauser — APPROVED** [review]

**Comment by pchaigno** [comment]
Reviews are in. All affected test suites passed except for k8s-1.20-kernel-5.4 which failed with known flake https://github.com/cilium/cilium/issues/16852. Marking ready to merge and merging.

**Pull request merged**